coursefunctions.h

```diff
@@ -529,33 +529,50 @@ void TestVertexShader(Buffer2D<PIXEL> & target)
 
         PIXEL colors[3] = {0xffff0000, 0xff00ff00, 0xff0000ff};
         // Your code for 'colorAttributes' goes here
+        colorAttributes[0].value.push_back(1.0);
+       	colorAttributes[0].value.push_back(0.0);
+	colorAttributes[0].value.push_back(0.0);
+	colorAttributes[1].value.push_back(0.0);
+	colorAttributes[1].value.push_back(1.0);
+	colorAttributes[1].value.push_back(0.0);
+	colorAttributes[2].value.push_back(0.0);
+	colorAttributes[2].value.push_back(0.0);
+	colorAttributes[2].value.push_back(1.0);
 
         FragmentShader myColorFragShader;
+	myColorFragShader.FragShader = ColorFragShader;
 
         Attributes colorUniforms;
         // Your code for the uniform goes here, if any (don't pass NULL here)
-        
+        Transform::reset(&colorUniforms);
+
         VertexShader myColorVertexShader;
-        // Your code for the vertex shader goes here 
+        myColorVertexShader.VertShader = VertShader;
 
         /******************************************************************
         * TRANSLATE (move +100 in the X direction, +50 in the Y direction)
         *****************************************************************/
         // Your translating code that integrates with 'colorUniforms', used by 'myColorVertexShader' goes here
+        Transform::translate(100, 50, 0, &colorUniforms);
```

Comment:
What you're doing is a smart idea: to pass the colorUniforms by reference.  I presume this updates the colorUniforms attribute with the new transformation.